### PR TITLE
Add public clients generation

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: src/AutoRest.CSharp.V3/Output/Models/Types/EnumType.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ private EnumType(Schema schema, BuildContext context, IEnumerable<ChoiceValue> c
             Declaration = BuilderHelpers.CreateTypeAttributes(
                 schema.CSharpName(),
                 $"{context.DefaultNamespace}.Models",
-                Accessibility.Public,
+                "public",
```
Comment (MiYanni) [inline src/AutoRest.CSharp.V3/Output/Models/Types/EnumType.cs]: Are we no longer using the Accessibility enum?

File: src/AutoRest.CSharp.V3/Output/Models/Client.cs [hunk 2]
```diff
@@ -1,31 +1,30 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
+// Licensed under the MIT License.
```
Comment (MiYanni) [inline src/AutoRest.CSharp.V3/Output/Models/Client.cs]: I noticed some of these file headers had been changed. Why would we remove the "See License.txt" part?

File: src/assets/Generator.Shared/CodeGenSchemaMemberAttribute.cs [hunk 3]
```diff
@@ -8,11 +8,8 @@ namespace Azure.Core
     [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
     internal class CodeGenSchemaMemberAttribute : Attribute
     {
-        public string SchemaName { get; }
```
Comment (MiYanni) [inline src/assets/Generator.Shared/CodeGenSchemaMemberAttribute.cs]: Why the removal of this property, but not the parameter in the constructor?

File: src/AutoRest.CSharp.V3/Generation/Writers/ClientWriter.cs [hunk 4]
```diff
@@ -300,26 +178,26 @@ private void WritePagingOperation(CodeWriter writer, PagingInfo pagingMethod, bo
...
                 var nullableInt = new CSharpType(typeof(int), true);
 
                 var continuationTokenText = pagingMethod.NextLinkName != null ? $"response.Value.{pagingMethod.NextLinkName}" : "null";
-                var asyncText = async ? "async " : string.Empty;
-                var awaitText = async ? "await " : string.Empty;
+                var asyncText = async ? "async" : string.Empty;
+                var awaitText = async ? "await" : string.Empty;
                 var configureAwaitText = async ? ".ConfigureAwait(false)" : string.Empty;
-                using (writer.Scope($"{asyncText}{funcType} FirstPageFunc({nullableInt} pageSizeHint)"))
+                using (writer.Scope($"{asyncText} {funcType} FirstPageFunc({nullableInt} pageSizeHint)"))
```
[the remark this answers is not in the document]
Comment (pakrym) [reply src/AutoRest.CSharp.V3/Generation/Writers/ClientWriter.cs]: Yep, Roslyn is good at removing double spaces.

File: src/assets/Generator.Shared/CodeGenSchemaMemberAttribute.cs (hunk 3 above)
Comment (pakrym) [reply src/assets/Generator.Shared/CodeGenSchemaMemberAttribute.cs]: We never read the value through the property, the way we use roslyn allows us to read the parameter value directly.

File: src/AutoRest.CSharp.V3/Output/Models/Client.cs (hunk 2 above)
Comment (pakrym) [reply src/AutoRest.CSharp.V3/Output/Models/Client.cs]: This is the correct header and it got applied to the new files I created.

File: src/AutoRest.CSharp.V3/Output/Models/Types/EnumType.cs (hunk 1 above)
Comment (pakrym) [reply src/AutoRest.CSharp.V3/Output/Models/Types/EnumType.cs]: I don't think it was adding anything.

**Comment by pakrym** [comment]
I noticed we are naming the operation group classes as Client now instead of Operations. Does that mean we intend to have every operation group as a separate client? Originally, I made operation groups as separate classes to simply keep similar operations together. They weren't initially supposed to become clients themselves.

Ha! I didn't realize that but I think I'll keep it :) We can always merge OGs if we get an ask.

**Pull request merged**